ambry-server/src/integration-test/java/com.github.ambry.server/MockCluster.java

```diff
@@ -221,8 +251,44 @@ private void initializeServer(DataNodeId dataNodeId, Properties sslProperties, b
     props.setProperty("replication.intra.replica.thread.throttle.sleep.duration.ms", "100");
     props.setProperty("replication.inter.replica.thread.throttle.sleep.duration.ms", "100");
     props.putAll(sslProperties);
-    VerifiableProperties propverify = new VerifiableProperties(props);
-    AmbryServer server = new AmbryServer(propverify, mockClusterAgentsFactory, notificationSystem, time);
+    return new VerifiableProperties(props);
+  }
+
+  /**
+   * Initialize {@link AmbryServer} node.
+   * @param dataNodeId {@link DataNodeId} object of the server initialized.
+   * @param sslProperties {@link Properties} object.
+   * @param enableHardDeletes {@code enableHardDeletes} flag.
+   * @param enableDataPrefetch {@code enableDataPrefetch} flag.
+   * @param notificationSystem {@link NotificationSystem} object.
+   * @param time {@link Time} object.
+   * @return {@link VerifiableProperties} object.
+   */
+  public void initializeServer(DataNodeId dataNodeId, Properties sslProperties, boolean enableHardDeletes,
+      boolean enableDataPrefetch, NotificationSystem notificationSystem, Time time) {
+    AmbryServer server =
+        new AmbryServer(createInitProperties(dataNodeId, enableDataPrefetch, enableHardDeletes, sslProperties),
+            mockClusterAgentsFactory, mockClusterSpectatorFactory, notificationSystem, time);
+    serverList.add(server);
+  }
+
+  /**
+   * Initialize {@link AmbryServer} node.
+   * @param dataNodeId {@link DataNodeId} object of the server initialized.
+   * @param sslProperties {@link Properties} object.
+   * @param enableHardDeletes {@code enableHardDeletes} flag.
+   * @param enableDataPrefetch {@code enableDataPrefetch} flag.
+   * @param notificationSystem {@link NotificationSystem} object.
+   * @param time {@link Time} object.
+   * @param mockClusterAgentsFactory {@link MockClusterAgentsFactory} object.
+   * @return {@link VerifiableProperties} object.
+   */
+  public void initializeServer(DataNodeId dataNodeId, Properties sslProperties, boolean enableHardDeletes,
+      boolean enableDataPrefetch, NotificationSystem notificationSystem, Time time,
+      MockClusterAgentsFactory mockClusterAgentsFactory) {
```

Comment:
I don't think we need two constructors here since they both construct AmbryServer the same way.  Just have one that takes the MockClusterAgentsFactory arg, and if null is passed, use the class default.